.github/workflows/build-java-v2.yaml

```diff
@@ -235,22 +205,76 @@ jobs:
           ecr-registry-code: ${{ needs.init.outputs.ecr-registry-code }}
           metadata: ${{ needs.init.outputs.metadata }}
 
-  run-e2e-test:
-    needs: [ update-overlays ]
-    if: ${{ needs.init.outputs.update-kustomize == 'true' && github.workflow != 'master' }}
+  # TODO: Check image status on TEST system
+
+  metadata:
+    needs: [ init, build, update-overlays ]
     runs-on: [ "self-hosted", "small-builder" ]
+
+    outputs:
+      metadata: ${{ steps.e2e-metadata.outputs.metadata }}
+
     steps:
-      - name: Generate e2e test metadata
+      - name: Generate e2e tests metadata
         id: e2e-metadata
-        uses: hawk-ai-aml/github-actions/e2e-test-metadata@master
+        uses: hawk-ai-aml/github-actions/e2e-tests-metadata@master
         with:
           repository-access-token: ${{ secrets.REPO_ACCESS_PAT }}
 
-      - name: Run e2e test
-        env:
-          USER: ${{ secrets.RECREATE_DEVELOP_GITHUB_USER }}
-          TOKEN: ${{ secrets.REPO_ACCESS_PAT }}
-          METADATA: ${{ steps.e2e-metadata.outputs.metadata }}
+  e2e-tests:
+    needs: metadata
+    runs-on: [ "self-hosted", "builder" ]
+    strategy:
+      fail-fast: false
+      matrix:
+        test: [ "amlcheck", "amnis", "bbamericas", "blacklist", "config", "configui", "countryvolumerules", "crosscard", "crr", "devportal", "dockfinancial", "effortful", "frontend", "grouping", "machineintelligence", "mapfintech", "passthrough", "pepsanction", "ratepay", "riskywords", "unusualfrequency", "volumechanges", "volumerisk", "worldline" ]
```

Comment:
will these stay hard-coded or do we provide a way to make them variable?

Reply:
Will be considered when we can finish all builds (build-java-multi, build-python, ...)